### DISCUSSION TITLE
Impact of Console React Components being deprecated

### DISCUSSION THREAD

**Discussion by apombo-ux** [twilio-labs/paste#2214] [Q&A] [answered]
As announced at the end of last week, the old Console React Components/ Library are now deprecated, but what does that mean, practically?

I did find definitions for deprecation in this older document [Sunsetting and Deprecating Console React & Console Sketch Components
](https://docs.google.com/spreadsheets/d/1ZPqOmLxAYkCBdhfGPRWNMvVAS4bYmcXmHok_1JeVCo4/edit#gid=0), which says that when component is deprecated it will no longer be updated for improvements or bugfixes and teams are strongly encouraged to switch over to the corresponding Paste component. Is that still accurate and complete? 

We're still in the process of migrating off Paste and I specifically want to confirm that while official support for the components may have ended, we won't lose access to the component packages yet.

Thanks!

**Comment by SiTaggart** [comment, marked as answer]
Hi @apombo-ux 

You are correct. Whilst the legacy Console React Components are deprecated, we are not blocking teams from deploying with them right now.

To reiterate something you call out though, it should be noted that Legacy Console React Components are no longer supported, will not be updated, and will not get any critical updates. In using them, your team is in effect assuming responsibility of shipping potentially harmful or insecure features to our customers.

Whilst we aren't blocking deployment _yet_, it should also be noted that teams are expected to migrate to the new Console in 2022. You will not be able to install legacy console react components in the new Console.

We do also track usage, installs and individual component usage in files. If we don't see the progress we'd like to see in reduction of legacy console react component usage, we will be exploring end of life options, potentially blocking deployment in the future. We're hopeful we don't need to do that, and teams will do the right thing in their migration to the new Console.

We look forward to seeing a world where no Twilio product is using these deprecated, legacy UI features, and welcome your support in making this happen!
